superset/sql_lab.py

```diff
@@ -132,7 +132,8 @@ def session_scope(nullpool: bool) -> Iterator[Session]:
         )
     if nullpool:
         engine = sqlalchemy.create_engine(database_uri, poolclass=NullPool)
-        session_class = sessionmaker(bind=engine)
+        session_class = sessionmaker()
```

Comment:
@etr2460 this was the only thing which I wonder was wrong. In the [docs](https://docs.sqlalchemy.org/en/13/orm/session_api.html#session-and-sessionmaker) where they recommend:

> The class also includes a method configure(), which can be used to specify additional keyword arguments to the factory, which will take effect for subsequent Session objects generated. This is usually used to associate one or more Engine objects with an existing sessionmaker factory before it is first used:
> 
> ```
> # application starts
> Session = sessionmaker()
>
> # ... later
> engine = create_engine('sqlite:///foo.db')
> Session.configure(bind=engine)
> ```

however we're not doing anything between creating and configuring the session.

Reply:
I don't believe this code is even run when trying to save a query. the session used here is only for running the query i thought?